docs/cub/api_docs/device_wide.rst

```diff
@@ -11,6 +11,72 @@ Device-Wide Primitives
    ../api/device
 
 
+.. _device-temp-storage:
+
+Determining Temporary Storage Requirements
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+**Two-Phase API** (Traditional)
+
+Most CUB device-wide algorithms follow a two-phase usage pattern:
+
+1. **Query Phase**: Call the algorithm with ``d_temp_storage = nullptr`` to determine required temporary storage size
+2. **Execution Phase**: Allocate storage and call the algorithm again to perform the actual operation
+
+**What arguments are needed during the query phase?**
+
+* **Template instantiation**: The query call must use the same template arguments as the execution call; only ``d_temp_storage``, ``temp_storage_bytes``, and problem-size parameters (for example ``num_items``) differ.
```

Comment:
Critical: I believe this is not correct. All template arguments must be the same. `d_temp_storage` and `temp_storage_bytes` have fixed types, so we can remove them here. The type for `num_items` must be the same, since we may use more temporary storage for 64-bit sizes vs. 32-bit sizes.

```suggestion
* **Template instantiation**: The query call must use the same template arguments as the execution call.
```